### ISSUE TITLE
Freezing when loading WAV files

### ISSUE THREAD

**Issue by kieranjol** [cth103/dcpomatic#1]
Hi - I'm using DCP-O-Matic v 2.12.20 git d4f07d4c25 to create DCP from DCDM. Windows 7 professional, Dual Xeon X5650 @2.67GHz (2 processors), 48 gig of ram, 7200rpm 10TB drive for input and output.

I find that loading an image sequence of tiffs takes almost no time at all, but when I import a folder of WAV files (usually six in a folder per reel), the program freezes for several minutes, I guess as it is examining the content. It's a little easier to drag and drop the files from Windows Explorer into the Content section as the program does not seem to freeze 100% - even though it stalls, it does allow me to drag and drop more files. Once I click anywhere on the GUI, I get the standard Windows 'Not Responding' reaction.
This makes it hard to build a DCP as the DCDM audio is usually split up into a folder per reel, with 6 wavs per folder. So I usually have to add the six wavs, wait a few mins, then re-assign the timing so that they start at the correct reel which then produces another long freeze, then do the same for the next four or so reels.

On a side note - i love how your software auto-assigns the single WAV files to the correct channels, i'm assuming based on the filename. And the fact that encoding uses every core on our Dual-Xeon CPU machine is so great :)

**Comment by cth103** [comment]
Hi Kieran,
That's strange!  Do you see a set of jobs added to the area underneath the preview when you start the import of your WAV files?

**Comment by cth103** [comment]
Also, are the WAVs coming from a local drive?

**Comment by kieranjol** [comment]
Yes to both! All coming from local drive and analysis jobs start after the
import...

**Comment by cth103** [comment]
...but the analysis jobs get stuck with no progress...?

**Comment by kieranjol** [comment]
Ok, so I have access to a test system again. I think I can now make a more accurate report.
I was unable to replicate the issue when adding the audio files before the image sequence. When I just add the audio files first, I see the 'examine content' bar progress as usual and it computes the digests and all is well.

I then replicated my usual behaviour which is to:
*  add the image sequence first, which does not result in the examine content bar proressing or anything, it just stalls for a few seconds and then seems to complete.
* Then add he audio files via add folder'
* This then causes the stall as previously reported, but eventually when DCP-o-matic un-stalls, I see that 'examine content' is OK and it ran for 2 seconds on each WAV.

**Comment by cth103** [comment]
Thanks for the detailed report! 

Is there any way you can try this with the current test version?  

https://dcpomatic.com/test-download

My tests suggest that it is better there; it would be great if you could test.

**Comment by kieranjol** [comment]
Hi Carl, sorry for the delay and thanks for the speedy fix. I think that the behaviour is indeed better. There is still a stall when adding audio after the image sequence, which gets more severe with each reel.
For example if I add one reel of an image sequence for a feature, then add the 6 channels of audio WAVs for that reel, I get a freeze of about 12 seconds which is fine. When I add a second image reel and then the next reel of audio WAVs, then the stall is about 25 seconds, and so on until reel four and five produce a stall of about 1 minute each when adding the audio.

I still think that this is quicker than it was before though.
